src/row.rs

```diff
@@ -176,13 +194,35 @@ impl Row {
     /// Draw the row and write the result to a buffer. An `offset` can be given, as well as a limit
     /// on the length of the row (`max_len`). After writing the characters, clear the rest of the
     /// line and move the cursor to the start of the next line.
-    pub fn draw(&self, offset: usize, max_len: usize, buffer: &mut String) -> Result<(), Error> {
+    pub fn draw(
+        &self, offset: usize, non_printable: bool, max_len: usize, buffer: &mut String,
+    ) -> Result<(), Error> {
         let mut current_hl_type = HlType::Normal;
         let chars = self.render.chars().skip(offset).take(max_len);
         let mut rx = self.render.chars().take(offset).map(|c| c.width().unwrap_or(1)).sum();
+        let mut rendered_char;
         for (c, mut hl_type) in chars.zip(self.hl.iter().skip(offset)) {
             if c.is_ascii_control() {
-                let rendered_char = if (c as u8) <= 26 { (b'@' + c as u8) as char } else { '?' };
+                if !non_printable {
+                    if (c as u8) <= 26 {
+                        rendered_char = (b'@' + c as u8) as char
+                    } else {
+                        rendered_char = '?'
+                    }
+                } else {
+                    match c {
+                        '\x0D' => rendered_char = '␍',
+                        // null
+                        '\x00' => rendered_char = '␀',
+                        // bell
+                        '\x07' => rendered_char = '␇',
+                        // backspace
+                        '\x08' => rendered_char = '␈',
+                        // escape
+                        '\x1B' => rendered_char = '␛',
+                        _ => rendered_char = '?',
```

Comment:
What about using a constant `HashMap`?

Reply:
I tried to use it, but it doesn't compile as all methods for creating a HashMap are not `const` and other options like a sorted Map of characters would add too many lines.